File: miscellaneous/sorting/selection-sort/20230101/selection_sort.py
from typing import List


def selection_sort(nums: List[int]) -> List[int]:
    is_sorted = True

    for i in range(len(nums) - 1):
        smallest = i

        for j in range(i + 1, len(nums)):
            if nums[j] < nums[smallest]:
                is_sorted = False
                smallest = j

        if is_sorted:
            return nums

        if smallest != i:
            [nums[i], nums[smallest]] = [nums[smallest], nums[i]]

    return nums

File: miscellaneous/sorting/selection-sort/20230101/selection-sort.php
<?php

declare(strict_types=1);

function selectionSort(array $nums): array
{
    $sorted = true;

    for ($i = 0; $i < count($nums) - 1; ++$i) {
        $min = $i;

        for ($j = $i + 1; $j < count($nums); ++$j) {
            if ($nums[$j] < $nums[$min]) {
                $sorted = false;
                $min = $j;
            }
        }

        $i != $min && [$nums[$i], $nums[$min]] = [$nums[$min], $nums[$i]];
    }

    return $nums;
}